ledger/complete/ledger.go

```diff
@@ -155,16 +155,23 @@ func (l *Ledger) Set(update *ledger.Update) (newState ledger.State, err error) {
 	l.metrics.UpdateCount()
 	l.metrics.UpdateValuesNumber(uint64(len(trieUpdate.Paths)))
 
-	err = l.wal.RecordUpdate(trieUpdate)
-	if err != nil {
-		return nil, fmt.Errorf("cannot update state, error while writing LedgerWAL: %w", err)
-	}
+	walChan := make(chan error)
+
+	go func() {
+		walChan <- l.wal.RecordUpdate(trieUpdate)
+		close(walChan)
```

Comment:
this assume that record update is always slower than the update operation, right? shall we close the walChan after the read at line 170 ?

Reply:
> this assume that record update is always slower than the update operation, right? shall we close the walChan after the read at line 170 ? otherwise we might reading a value from a closed channel.

Good spot, no, we shouldn't close this channel at all. Fixed.